src/server/mappings/lib/get_root_properties_objects.js

```diff
@@ -0,0 +1,28 @@
+import { getRootProperties } from './get_root_properties';
+
+/**
+ *  Get the property mappings for the root type in the EsMappingsDsl
+ *  where the properties are objects
+ *
+ *  If the mappings don't have a root type, or the root type is not
+ *  an object type (it's a keyword or something) this function will
+ *  throw an error.
+ *
+ *  This data can be found at `{indexName}.mappings.{typeName}.properties`
+ *  in the es indices.get() response where the properties are objects.
+ *
+ *  @param  {EsMappingsDsl} mappings
+ *  @return {EsPropertyMappings}
+ */
+export function getRootPropertiesObjects(mappings) {
+  const rootProperties = getRootProperties(mappings);
+  return Object.entries(rootProperties).reduce((acc, [key, value]) => {
+
+    // we consider the existence of the properties or type of object to designate that this is an object
```

Comment:
I think this should say "... to designate that this is a type"

Reply:
hm. I guess we are calling these "root properties objects", so maybe type makes sense? But to my personally, the root properties object is the same thing as a top level type. Maybe saying "... to designate that this is a top level mapping type"

Reply:
Yeah, the naming is quite weird. I was trying to keep the way we were parsing the mappings separate from their usage within Kibana's saved objects, which prevented me from naming it `getSavedObjectTypes`. Perhaps going with a long and somewhat clumsy name of `getRootPropertiesThatAreObjectDatatypes` is better?

Reply:
My comment wasn't talking about saved objects in particular. I was thinking of the elasticsearch naming, of "top level mapping type" which seems like a common way to describe that thing.

```
mappings
|-type (_doc)
  |-properties
    |-object datatype (kibana saved object type, sub-type of doc)
    |-field (fields common to all sibling object datatypes)
    |- ...
```

But ah, there's always the top level `_doc` type, so everything below it is just some _thing_, which we call types in Kibana, but which are generically `Object Datatype`s.

Does that above diagram describe it?

I now think the comment should say ".. that this is an object datatype"